src/Http/ControllerFactory.php

```diff
@@ -46,22 +42,20 @@ public function create(ServerRequest $request, Response $response)
             $this->missingController($request);
         }
 
-        return $reflection->newInstance($request, $response, $controller);
+        return $reflection->newInstance($request, $response);
```

Comment:
This should be safe. The controller uses the request params as a fallback for the name.